util/build_configs/cray-internal/setenv-shasta-x86_64.bash

```diff
@@ -0,0 +1,503 @@
+    ( * )
+        log_info "NO building Chapel component: clean"
+        ;;
+    esac
+
+    log_info "End $setenv"
+else
+
+# ===============
+# setenv callback
+# ===============
+
+    setenv=$( basename "${BASH_SOURCE[0]}" )
+    cwd=$( cd $(dirname "${BASH_SOURCE[0]}") && pwd )
+    source $cwd/../functions.bash
+
+    log_info "Begin callback $setenv"
+    log_debug "with config=$BUILD_CONFIGS_CALLBACK"
+
+    # Exit immediately to skip (avoid building) unwanted Chapel configs
+
+    if [ "$CHPL_LIB_PIC" == pic ]; then
+      # skip Chapel make for any communication and launcher that are not none
+      # because pic support is for python interoperability which
+      # only runs on the login node at the moment.
+      if [ "$CHPL_COMM" != none ]; then
+        log_info "Skip Chapel make for libpic=$CHPL_LIB_PIC, comm=$CHPL_COMM"
+        exit 0
+      fi
+      if [ "$CHPL_LAUNCHER" != none ]; then
+        log_info "Skip Chapel make for libpic=$CHPL_LIB_PIC, launcher=$CHPL_LAUNCHER"
+        exit 0
+      fi
+    fi
+
+    if [ "$CHPL_COMM" == ofi ]; then
+        if [ "$CHPL_LAUNCHER" == none ]; then
+
+            # skip Chapel make because comm == ofi needs a Chapel launcher
+            log_info "Skip Chapel make for comm=$CHPL_COMM, launcher=$CHPL_LAUNCHER"
+            exit 0
+        fi
+        if [ "$CHPL_COMM_SUBSTRATE" != none ]; then
+
+            # skip Chapel make because comm == ofi does not take a substrate
+            log_info "Skip Chapel make for comm=$CHPL_COMM, substrate=$CHPL_COMM_SUBSTRATE"
+            exit 0
+        fi
+        log_info "Unset CHPL_COMM_SUBSTRATE for comm=$CHPL_COMM, substrate=$CHPL_COMM_SUBSTRATE"
+        unset CHPL_COMM_SUBSTRATE
+    elif [ "$CHPL_COMM" == gasnet ]; then
+        if [ "$CHPL_COMM_SUBSTRATE" == none ]; then
+
+            # skip Chapel make because comm == gasnet needs a substrate
+            log_info "Skip Chapel make for comm=$CHPL_COMM, substrate=$CHPL_COMM_SUBSTRATE"
+            exit 0
+        fi
+    elif [ "$CHPL_COMM" == none ]; then
+        if [ "$CHPL_COMM_SUBSTRATE" != none ]; then
+
+            # skip Chapel make because we already built the runtime with comm=none
+            log_info "Skip Chapel make for comm=$CHPL_COMM, substrate=$CHPL_COMM_SUBSTRATE"
+            exit 0
+        fi
+        log_info "Unset CHPL_COMM_SUBSTRATE for comm=$CHPL_COMM, substrate=$CHPL_COMM_SUBSTRATE"
+        unset CHPL_COMM_SUBSTRATE
+    else
+        log_error "unrecognized CHPL_COMM=$CHPL_COMM"
+        exit 2
+    fi
+
+# =========================
+# setenv callback functions
+# =========================
+
+    source $cwd/../module-functions.bash
+
+    # check module system
+    ck_module_list
+
+    if [ -n "$verbose" ] ; then
+        log_debug "Module list (sorted), after build_configs, before setenv:"
+        list_loaded_modules
+    fi
+
+    # Please keep the gen versions in compiler_versions.bash the same as these!
+    #[TODO] gen_version_gcc=7.3.0
```

Comment:
Are we just using the system default gcc?

Reply:
Yes, for now.